rackcity/views/change_plan_views.py

```diff
@@ -0,0 +1,294 @@
+                "errors": "Must include 'id' when deleting an asset"
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    id = data['id']
+    try:
+        existing_change_plan = ChangePlan.objects.get(id=id)
+
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    "Model" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing asset with id="+str(id)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    try:
+        existing_change_plan.delete()
+    except Exception as error:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.DELETE_ERROR.value +
+                    "Change Plan" +
+                    GenericFailure.ON_DELETE.value,
+                "errors": str(error)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    return JsonResponse(
+        {
+            "success_message":
+                Status.SUCCESS.value +
+                "Change Plan " + str(existing_change_plan.name) + " deleted"
+        },
+        status=HTTPStatus.OK
+    )
+
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def change_plan_modify(request):
+    """
+    Modify single existing change plan
+    """
+    data = JSONParser().parse(request)
+    if 'id' not in data:
+        return JsonResponse(
+            {
+                "failure_message":
+                Status.MODIFY_ERROR.value + GenericFailure.INTERNAL.value,
+                    "errors": "Must include 'id' when modifying an asset"
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    id = data['id']
+    try:
+        existing_change_plan = ChangePlan.objects.get(id=id)
+
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    "Model" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing asset with id="+str(id)
+            },
+            status=HTTPStatus.BAD_REQUEST
+        )
+    for field in data.keys():
+        value = data[field]
+        setattr(existing_change_plan, field, value)
+    try:
+        existing_change_plan.save()
+        return JsonResponse(
+            {
+                "success_message":
+                    Status.SUCCESS.value +
+                    "Change Plan " +
+                    str(existing_change_plan.name) +
+                    " modified",
+                "related_id": str(existing_change_plan.id)
+            },
+            status=HTTPStatus.OK,
+        )
+    except Exception as error:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    parse_save_validation_error(error, "Asset"),
```

Comment:
copy change typo - "Asset" is used in this method `parse_save_validation_error` just to use the string in the error message that's generated, should be "Change Plan" or whatever you want to call it